server/src/main/java/org/elasticsearch/index/snapshots/blobstore/SlicedInputStream.java

```diff
@@ -40,7 +40,15 @@ protected SlicedInputStream(final int numSlices) {
     }
 
     private InputStream nextStream() throws IOException {
-        assert initialized == false || currentStream != null;
+        assert initialized == false || currentStream != null || (markedSlice >= 0 && markedSliceOffset >= 0)
```

Comment:
I don't think this weakening makes sense for all the other callers of `nextStream()` that aren't resetting back to a marked position.

Basically I don't think `nextStream()` really makes sense in the context of `reset()`. We're not moving to the _next_ stream, we're moving to some arbitrary earlier stream. Setting things up in `reset()` so that `nextStream()` does the right thing seems way more complex than just moving to the right stream directly.

Reply:
Sounds good, then made reset() set the currentStream directly.